This is a collection of patches that deactivates or increase the error threshold of flaky tests.
Excepted for the first one, they are derived from the following LAMMPS PRs:
  https://github.com/lammps/lammps/pull/4349
  https://github.com/lammps/lammps/pull/4439

diff --git a/unittest/force-styles/tests/dihedral-table_spline.yaml b/unittest/force-styles/tests/dihedral-table_spline.yaml
index 10e56e4a6d..e27ad71d78 100644
--- a/unittest/force-styles/tests/dihedral-table_spline.yaml
+++ b/unittest/force-styles/tests/dihedral-table_spline.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 22 Dec 2022
 date_generated: Mon Dec 26 16:49:37 2022
-epsilon: 1.0e-13
+epsilon: 5.0e-13
 skip_tests:
 prerequisites: ! |
   atom full

From cdd538e24efbfcd0e656e5b66d8dce11243ab5a0 Mon Sep 17 00:00:00 2001
From: Axel Kohlmeyer <akohlmey@gmail.com>
Date: Sat, 5 Oct 2024 01:57:56 -0400
Subject: [PATCH 05/41] relax epsilon for IntelLLVM compiler

---
 unittest/force-styles/tests/atomic-pair-meam_2nn.yaml           | 2 +-
 unittest/force-styles/tests/atomic-pair-polymorphic_eam.yaml    | 2 +-
 unittest/force-styles/tests/atomic-pair-reaxff_lgvdw.yaml       | 2 +-
 .../force-styles/tests/dihedral-cosine_squared_restricted.yaml  | 2 +-
 unittest/force-styles/tests/fix-timestep-addtorque_const.yaml   | 2 +-
 unittest/force-styles/tests/fix-timestep-gravity.yaml           | 2 +-
 unittest/force-styles/tests/fix-timestep-heat.yaml              | 2 +-
 unittest/force-styles/tests/fix-timestep-shake_bond.yaml        | 2 +-
 unittest/force-styles/tests/fix-timestep-smd_couple.yaml        | 2 +-
 unittest/force-styles/tests/fix-timestep-spring_tether.yaml     | 2 +-
 unittest/force-styles/tests/fix-timestep-wall_lj126_const.yaml  | 2 +-
 unittest/force-styles/tests/kspace-ewald_conp_charge.yaml       | 2 +-
 unittest/force-styles/tests/manybody-pair-uf3.yaml              | 2 +-
 unittest/force-styles/tests/mol-pair-coul_diel.yaml             | 2 +-
 14 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/unittest/force-styles/tests/atomic-pair-meam_2nn.yaml b/unittest/force-styles/tests/atomic-pair-meam_2nn.yaml
index 0fd77d67ba1..4eb285fa73d 100644
--- a/unittest/force-styles/tests/atomic-pair-meam_2nn.yaml
+++ b/unittest/force-styles/tests/atomic-pair-meam_2nn.yaml
@@ -2,7 +2,7 @@
 lammps_version: 27 Jun 2024
 tags: slow
 date_generated: Mon Aug 12 23:46:29 2024
-epsilon: 7.5e-12
+epsilon: 2e-11
 skip_tests:
 prerequisites: ! |
   pair meam
diff --git a/unittest/force-styles/tests/atomic-pair-polymorphic_eam.yaml b/unittest/force-styles/tests/atomic-pair-polymorphic_eam.yaml
index e8a29f76bae..b97c1fad6f3 100644
--- a/unittest/force-styles/tests/atomic-pair-polymorphic_eam.yaml
+++ b/unittest/force-styles/tests/atomic-pair-polymorphic_eam.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:38 2022
-epsilon: 7.5e-14
+epsilon: 1e-13
 skip_tests:
 prerequisites: ! |
   pair polymorphic
diff --git a/unittest/force-styles/tests/atomic-pair-reaxff_lgvdw.yaml b/unittest/force-styles/tests/atomic-pair-reaxff_lgvdw.yaml
index 363f7cf74d3..3ccc44fe4bd 100644
--- a/unittest/force-styles/tests/atomic-pair-reaxff_lgvdw.yaml
+++ b/unittest/force-styles/tests/atomic-pair-reaxff_lgvdw.yaml
@@ -2,7 +2,7 @@
 lammps_version: 30 Jul 2021
 tags: slow, unstable
 date_generated: Mon Aug 23 20:32:03 2021
-epsilon: 4e-12
+epsilon: 1e-11
 skip_tests:
 prerequisites: ! |
   pair reaxff
diff --git a/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml b/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
index 3f4d217b9a3..c2c2b8cc6ba 100644
--- a/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
+++ b/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
@@ -2,7 +2,7 @@
 lammps_version: 7 Feb 2024
 tags:
 date_generated: Sat Apr 13 11:41:16 2024
-epsilon: 1.0e-11
+epsilon: 5.0e-11
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-addtorque_const.yaml b/unittest/force-styles/tests/fix-timestep-addtorque_const.yaml
index 4635ae54851..f239d0ba7cc 100644
--- a/unittest/force-styles/tests/fix-timestep-addtorque_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-addtorque_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:58 2022
-epsilon: 2.5e-11
+epsilon: 7.5e-11
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-heat.yaml b/unittest/force-styles/tests/fix-timestep-heat.yaml
index 80b679aa1f6..563ebf4d978 100644
--- a/unittest/force-styles/tests/fix-timestep-heat.yaml
+++ b/unittest/force-styles/tests/fix-timestep-heat.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:58 2022
-epsilon: 2e-14
+epsilon: 5e-14
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-shake_bond.yaml b/unittest/force-styles/tests/fix-timestep-shake_bond.yaml
index 49ac1c050af..62230508a47 100644
--- a/unittest/force-styles/tests/fix-timestep-shake_bond.yaml
+++ b/unittest/force-styles/tests/fix-timestep-shake_bond.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 5e-11
+epsilon: 2e-10
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-smd_couple.yaml b/unittest/force-styles/tests/fix-timestep-smd_couple.yaml
index 9b9c5d93bf6..3b8d74bffa3 100644
--- a/unittest/force-styles/tests/fix-timestep-smd_couple.yaml
+++ b/unittest/force-styles/tests/fix-timestep-smd_couple.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 2e-14
+epsilon: 1e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-spring_tether.yaml b/unittest/force-styles/tests/fix-timestep-spring_tether.yaml
index 4786031d3dd..99016d6b5d8 100644
--- a/unittest/force-styles/tests/fix-timestep-spring_tether.yaml
+++ b/unittest/force-styles/tests/fix-timestep-spring_tether.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 2e-14
+epsilon: 1e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_lj126_const.yaml b/unittest/force-styles/tests/fix-timestep-wall_lj126_const.yaml
index e0b8bef40fb..c7089a2592d 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_lj126_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_lj126_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 2e-14
+epsilon: 1e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/kspace-ewald_conp_charge.yaml b/unittest/force-styles/tests/kspace-ewald_conp_charge.yaml
index 9bc190a766e..bf6b922151c 100644
--- a/unittest/force-styles/tests/kspace-ewald_conp_charge.yaml
+++ b/unittest/force-styles/tests/kspace-ewald_conp_charge.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 7 Feb 2024
 date_generated: Mon Mar  4 09:44:30 2024
-epsilon: 1e-12
+epsilon: 2e-12
 skip_tests: gpu kokkos_omp omp
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/manybody-pair-uf3.yaml b/unittest/force-styles/tests/manybody-pair-uf3.yaml
index 534b489c782..c5fa99aa844 100644
--- a/unittest/force-styles/tests/manybody-pair-uf3.yaml
+++ b/unittest/force-styles/tests/manybody-pair-uf3.yaml
@@ -2,7 +2,7 @@
 lammps_version: 24 Mar 2022
 tags:
 date_generated: Tue Nov 28 14:51:03 2023
-epsilon: 1e-09
+epsilon: 2e-09
 skip_tests: 
 prerequisites: ! |
   pair uf3
diff --git a/unittest/force-styles/tests/mol-pair-coul_diel.yaml b/unittest/force-styles/tests/mol-pair-coul_diel.yaml
index ed63f961c73..9926d78d065 100644
--- a/unittest/force-styles/tests/mol-pair-coul_diel.yaml
+++ b/unittest/force-styles/tests/mol-pair-coul_diel.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:29 2022
-epsilon: 2e-13
+epsilon: 2.5e-13
 skip_tests:
 prerequisites: ! |
   atom full

From 248070c4b8d05c362f724b7f797b2ed9a6a0c322 Mon Sep 17 00:00:00 2001
From: Axel Kohlmeyer <akohlmey@gmail.com>
Date: Sat, 5 Oct 2024 02:40:42 -0400
Subject: [PATCH 06/41] use -fp-model precise for IntelLLVM to avoid
 miscompilation

---
 cmake/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index afae18745fe..356d98136ba 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -118,7 +118,7 @@ endif()
 
 # silence excessive warnings for new Intel Compilers
 if(CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
-  set(CMAKE_TUNE_DEFAULT "-Wno-tautological-constant-compare -Wno-unused-command-line-argument")
+  set(CMAKE_TUNE_DEFAULT "-fp-model precise -Wno-tautological-constant-compare -Wno-unused-command-line-argument")
 endif()
 
 # silence excessive warnings for PGI/NVHPC compilers

From ca6e9abefaa53ff48320ac750fa59a7ff07a38f1 Mon Sep 17 00:00:00 2001
From: Axel Kohlmeyer <akohlmey@gmail.com>
Date: Sat, 5 Oct 2024 03:37:33 -0400
Subject: [PATCH 08/41] update a few more tests for IntelLLVM

---
 unittest/force-styles/tests/manybody-pair-pace_product.yaml   | 2 +-
 unittest/force-styles/tests/manybody-pair-pace_recursive.yaml | 2 +-
 unittest/force-styles/tests/manybody-pair-tersoff.yaml        | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/unittest/force-styles/tests/manybody-pair-pace_product.yaml b/unittest/force-styles/tests/manybody-pair-pace_product.yaml
index b178586ab20..6db9f4220a5 100644
--- a/unittest/force-styles/tests/manybody-pair-pace_product.yaml
+++ b/unittest/force-styles/tests/manybody-pair-pace_product.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:48 2022
-epsilon: 5e-12
+epsilon: 7.5e-09
 skip_tests:
 prerequisites: ! |
   pair pace
diff --git a/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml b/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
index 6105debb671..61f7ce0ac95 100644
--- a/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
+++ b/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 10 Mar 2021
 date_generated: Wed Apr  7 19:30:07 2021
-epsilon: 5e-12
+epsilon: 7.5e-09
 prerequisites: ! |
   pair pace
 pre_commands: ! |
diff --git a/unittest/force-styles/tests/manybody-pair-tersoff.yaml b/unittest/force-styles/tests/manybody-pair-tersoff.yaml
index 1df552c0432..8b9adecbad3 100644
--- a/unittest/force-styles/tests/manybody-pair-tersoff.yaml
+++ b/unittest/force-styles/tests/manybody-pair-tersoff.yaml
@@ -2,7 +2,7 @@
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:50 2022
 epsilon: 2e-11
-skip_tests:
+skip_tests: intel
 prerequisites: ! |
   pair tersoff
 pre_commands: ! |

From 56cc9d03bdd35eaa23e0febffea6f40892a04d96 Mon Sep 17 00:00:00 2001
From: Axel Kohlmeyer <akohlmey@gmail.com>
Date: Thu, 10 Oct 2024 06:11:32 -0400
Subject: [PATCH 33/41] skip tests that are unstable on GPU

---
 unittest/force-styles/tests/mol-pair-lj_cut_coul_cut_soft.yaml  | 2 +-
 unittest/force-styles/tests/mol-pair-lj_cut_coul_long_soft.yaml | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/unittest/force-styles/tests/mol-pair-lj_cut_coul_cut_soft.yaml b/unittest/force-styles/tests/mol-pair-lj_cut_coul_cut_soft.yaml
index 485730531f5..2c54e2a0854 100644
--- a/unittest/force-styles/tests/mol-pair-lj_cut_coul_cut_soft.yaml
+++ b/unittest/force-styles/tests/mol-pair-lj_cut_coul_cut_soft.yaml
@@ -2,7 +2,7 @@
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:31 2022
 epsilon: 5e-13
-skip_tests:
+skip_tests: gpu
 prerequisites: ! |
   atom full
   pair lj/cut/coul/cut/soft
diff --git a/unittest/force-styles/tests/mol-pair-lj_cut_coul_long_soft.yaml b/unittest/force-styles/tests/mol-pair-lj_cut_coul_long_soft.yaml
index a1e89e54c0e..e3a2d378412 100644
--- a/unittest/force-styles/tests/mol-pair-lj_cut_coul_long_soft.yaml
+++ b/unittest/force-styles/tests/mol-pair-lj_cut_coul_long_soft.yaml
@@ -2,7 +2,7 @@
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:31 2022
 epsilon: 7.5e-12
-skip_tests:
+skip_tests: gpu
 prerequisites: ! |
   atom full
   pair lj/cut/coul/long/soft

From f10d1a30fc4fb32180370d9ff6673b3f2492da97 Mon Sep 17 00:00:00 2001
From: Axel Kohlmeyer <akohlmey@gmail.com>
Date: Tue, 21 Jan 2025 00:47:41 -0500
Subject: [PATCH 2/4] increase epsilon for passing tests on Linux ARM64

---
 unittest/force-styles/tests/atomic-pair-lepton_sphere.yaml     | 1 +
 unittest/force-styles/tests/atomic-pair-lj_cut_sphere.yaml     | 1 +
 unittest/force-styles/tests/atomic-pair-lj_expand_sphere.yaml  | 1 +
 unittest/force-styles/tests/bond-harmonic_restrain.yaml        | 2 +-
 .../force-styles/tests/dihedral-cosine_squared_restricted.yaml | 3 +--
 unittest/force-styles/tests/fix-timestep-recenter-coords.yaml  | 2 +-
 unittest/force-styles/tests/fix-timestep-recenter-init.yaml    | 2 +-
 unittest/force-styles/tests/fix-timestep-spring_rg.yaml        | 2 +-
 .../force-styles/tests/fix-timestep-wall_harmonic_const.yaml   | 2 +-
 .../force-styles/tests/fix-timestep-wall_lepton_const.yaml     | 2 +-
 unittest/force-styles/tests/fix-timestep-wall_lj93_const.yaml  | 2 +-
 unittest/force-styles/tests/fix-timestep-wall_morse_const.yaml | 2 +-
 .../force-styles/tests/fix-timestep-wall_table_linear.yaml     | 2 +-
 .../force-styles/tests/fix-timestep-wall_table_spline.yaml     | 2 +-
 unittest/force-styles/tests/manybody-pair-pace_product.yaml    | 2 +-
 unittest/force-styles/tests/manybody-pair-pace_recursive.yaml  | 2 +-
 unittest/force-styles/tests/mol-pair-lepton.yaml               | 2 +-
 17 files changed, 17 insertions(+), 15 deletions(-)

diff --git a/unittest/force-styles/tests/atomic-pair-lepton_sphere.yaml b/unittest/force-styles/tests/atomic-pair-lepton_sphere.yaml
index 222aa8b93c3..0e00e418a19 100644
--- a/unittest/force-styles/tests/atomic-pair-lepton_sphere.yaml
+++ b/unittest/force-styles/tests/atomic-pair-lepton_sphere.yaml
@@ -1,6 +1,7 @@
 ---
 lammps_version: 28 Mar 2023
 date_generated: Fri Apr  7 18:04:29 2023
+tags: unstable
 epsilon: 7.5e-13
 skip_tests: single
 prerequisites: ! |
diff --git a/unittest/force-styles/tests/atomic-pair-lj_cut_sphere.yaml b/unittest/force-styles/tests/atomic-pair-lj_cut_sphere.yaml
index 3a5122a896b..193a65122e7 100644
--- a/unittest/force-styles/tests/atomic-pair-lj_cut_sphere.yaml
+++ b/unittest/force-styles/tests/atomic-pair-lj_cut_sphere.yaml
@@ -1,6 +1,7 @@
 ---
 lammps_version: 28 Mar 2023
 date_generated: Thu Mar 30 14:38:22 2023
+tags: unstable
 epsilon: 7.5e-13
 skip_tests: single
 prerequisites: ! |
diff --git a/unittest/force-styles/tests/atomic-pair-lj_expand_sphere.yaml b/unittest/force-styles/tests/atomic-pair-lj_expand_sphere.yaml
index 24a17a275c8..d929544809d 100644
--- a/unittest/force-styles/tests/atomic-pair-lj_expand_sphere.yaml
+++ b/unittest/force-styles/tests/atomic-pair-lj_expand_sphere.yaml
@@ -1,6 +1,7 @@
 ---
 lammps_version: 28 Mar 2023
 date_generated: Fri Apr  7 18:07:13 2023
+tags: unstable
 epsilon: 7.5e-13
 skip_tests: single
 prerequisites: ! |
diff --git a/unittest/force-styles/tests/bond-harmonic_restrain.yaml b/unittest/force-styles/tests/bond-harmonic_restrain.yaml
index 07546775ab0..485dbfeafc7 100644
--- a/unittest/force-styles/tests/bond-harmonic_restrain.yaml
+++ b/unittest/force-styles/tests/bond-harmonic_restrain.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 8 Feb 2023
 date_generated: Tue Mar  7 21:07:27 2023
-epsilon: 2.5e-13
+epsilon: 5.0e-13
 skip_tests: extract
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml b/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
index c2c2b8cc6ba..0e43bc1741c 100644
--- a/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
+++ b/unittest/force-styles/tests/dihedral-cosine_squared_restricted.yaml
@@ -1,8 +1,7 @@
 ---
 lammps_version: 7 Feb 2024
-tags:
 date_generated: Sat Apr 13 11:41:16 2024
-epsilon: 5.0e-11
+epsilon: 2.0e-10
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-spring_rg.yaml b/unittest/force-styles/tests/fix-timestep-spring_rg.yaml
index a6c5844b6c6..bfd9310012a 100644
--- a/unittest/force-styles/tests/fix-timestep-spring_rg.yaml
+++ b/unittest/force-styles/tests/fix-timestep-spring_rg.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Thu Mar 17 19:43:17 2022
-epsilon: 2e-14
+epsilon: 5.0e-14
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_harmonic_const.yaml b/unittest/force-styles/tests/fix-timestep-wall_harmonic_const.yaml
index 57f38b4c37e..58065829295 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_harmonic_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_harmonic_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 4e-14
+epsilon: 5.0e-14
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_lepton_const.yaml b/unittest/force-styles/tests/fix-timestep-wall_lepton_const.yaml
index 947bc6a95ad..079383e04c6 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_lepton_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_lepton_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 8 Feb 2023
 date_generated: Thu Feb 23 00:40:51 2023
-epsilon: 4e-14
+epsilon: 5.0e-14
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_lj93_const.yaml b/unittest/force-styles/tests/fix-timestep-wall_lj93_const.yaml
index 5431a8e0a89..590ed1f103f 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_lj93_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_lj93_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 27 Jun 2024
 date_generated: Fri Mar 18 22:18:01 2022
-epsilon: 2e-14
+epsilon: 1.0e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_morse_const.yaml b/unittest/force-styles/tests/fix-timestep-wall_morse_const.yaml
index 391070609f9..08080a6274d 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_morse_const.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_morse_const.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 8 Feb 2023
 date_generated: Thu Feb 23 15:26:55 2023
-epsilon: 4e-14
+epsilon: 1.0e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_table_linear.yaml b/unittest/force-styles/tests/fix-timestep-wall_table_linear.yaml
index 6291de136af..ee860262167 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_table_linear.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_table_linear.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 8 Feb 2023
 date_generated: Thu Feb 23 00:56:30 2023
-epsilon: 4e-14
+epsilon: 2.0e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/fix-timestep-wall_table_spline.yaml b/unittest/force-styles/tests/fix-timestep-wall_table_spline.yaml
index 6c6c674342b..7f27c59e329 100644
--- a/unittest/force-styles/tests/fix-timestep-wall_table_spline.yaml
+++ b/unittest/force-styles/tests/fix-timestep-wall_table_spline.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 8 Feb 2023
 date_generated: Thu Feb 23 00:56:30 2023
-epsilon: 4e-14
+epsilon: 2.0e-13
 skip_tests:
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/manybody-pair-pace_product.yaml b/unittest/force-styles/tests/manybody-pair-pace_product.yaml
index 6db9f4220a5..fe4cde6dac6 100644
--- a/unittest/force-styles/tests/manybody-pair-pace_product.yaml
+++ b/unittest/force-styles/tests/manybody-pair-pace_product.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 17 Feb 2022
 date_generated: Fri Mar 18 22:17:48 2022
-epsilon: 7.5e-09
+epsilon: 1.5e-08
 skip_tests:
 prerequisites: ! |
   pair pace
diff --git a/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml b/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
index 61f7ce0ac95..3740718675b 100644
--- a/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
+++ b/unittest/force-styles/tests/manybody-pair-pace_recursive.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 10 Mar 2021
 date_generated: Wed Apr  7 19:30:07 2021
-epsilon: 7.5e-09
+epsilon: 1.5e-08
 prerequisites: ! |
   pair pace
 pre_commands: ! |
diff --git a/unittest/force-styles/tests/mol-pair-lepton.yaml b/unittest/force-styles/tests/mol-pair-lepton.yaml
index 33576e81c2a..c0e26b34f6b 100644
--- a/unittest/force-styles/tests/mol-pair-lepton.yaml
+++ b/unittest/force-styles/tests/mol-pair-lepton.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 21 Nov 2023
 date_generated: Thu Jan 18 11:01:50 2024
-epsilon: 5e-14
+epsilon: 1e-13
 skip_tests: intel
 prerequisites: ! |
   atom full
diff --git a/unittest/force-styles/tests/kspace-pppm_conp_charge.yaml b/unittest/force-styles/tests/kspace-pppm_conp_charge.yaml
index 38161ee044..ed41e8db52 100644
--- a/unittest/force-styles/tests/kspace-pppm_conp_charge.yaml
+++ b/unittest/force-styles/tests/kspace-pppm_conp_charge.yaml
@@ -1,7 +1,7 @@
 ---
 lammps_version: 7 Feb 2024
 date_generated: Mon Mar  4 09:44:31 2024
-epsilon: 3e-12
+epsilon: 9e-12
 skip_tests: gpu kokkos_omp omp
 prerequisites: ! |
   atom full
---
Support for mpi4py 4.x

diff --git a/unittest/python/python-open.py b/unittest/python/python-open.py
index f555be25dd..a1d5617986 100644
--- a/unittest/python/python-open.py
+++ b/unittest/python/python-open.py
@@ -7,8 +7,8 @@ has_mpi4py=False
 
 try:
     from mpi4py import __version__ as mpi4py_version
-    # tested to work with mpi4py versions 2 and 3
-    has_mpi4py = mpi4py_version.split('.')[0] in ['2','3']
+    # tested to work with mpi4py versions 2, 3, and 4
+    has_mpi4py = mpi4py_version.split('.')[0] in ['2','3','4']
 except:
     pass
